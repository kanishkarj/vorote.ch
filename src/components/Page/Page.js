import React, { useRef, useEffect } from 'react';
import styled from 'styled-components';
import { grid } from 'styled-components-grid';
import { mediaUp } from '../../utils/get-media-query';
import { margin, marginBottom } from '../../utils/get-margin';
import { lineHeight } from '../../utils/get-line-height';

const StyledPage = styled.div`
  ${marginBottom(2)}
  ${grid.unit({ size: { sm: 7 / 12, md: 2 / 3 } })}
`;

const PageInner = styled.div`
  padding: 25px 20px;
  ${mediaUp.sm`
    padding: 30px 20px;
  `}
  ${mediaUp.md`
    padding: 40px 35px;
  `}
`;

const Title = styled.h1`
  font-size: ${(props) => props.theme.fontSizes.base * 2.5}px;
  font-weight: ${(props) => props.theme.fontWeights.heading};
  ${lineHeight(2)}
  margin-top: 0px;
  ${marginBottom(1.45)}
`;

const PageContent = styled.div`
  font-size: ${(props) => props.theme.fontSizes.base}px;
  ${lineHeight(1)}
  ${margin(0, 0, 1)}
`;

type Props = {
  title?: string,
  children: React.Node
};

const Page = ({ title, children }: Props) => {
  const pageRef = useRef();

  useEffect(() => {
    pageRef.current.scrollIntoView();
  });

  return (
    <StyledPage ref={pageRef}>
      <PageInner>
        {title && <Title>{title}</Title>}
        <PageContent>
          {children}
        </PageContent>
      </PageInner>
    </StyledPage>
  );
};

export default Page;
