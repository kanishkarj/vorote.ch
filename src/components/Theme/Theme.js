import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
