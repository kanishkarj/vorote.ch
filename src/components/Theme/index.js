// @flow strict
export { default } from './Theme';
