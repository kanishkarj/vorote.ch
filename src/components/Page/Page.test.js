// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Page from './Page';
import 'jest-styled-components'
import Theme from '../Theme';

describe('Page', () => {
  const props = {
    children: 'test',
    title: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(
    <Theme>
      <Page {...props} />
    </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
