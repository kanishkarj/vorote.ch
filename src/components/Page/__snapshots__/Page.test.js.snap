// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
.c0 {
  margin-bottom: 52px;
  box-sizing: border-box;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
}

.c1 {
  padding: 25px 20px;
}

.c2 {
  font-size: 40px;
  font-weight: 600;
  line-height: 3.25em;
  margin-top: 0px;
  margin-bottom: 37.699999999999996px;
}

.c3 {
  font-size: 16px;
  line-height: 1.625em;
  margin: 0px 0px 26px;
}

@media (min-width:36em) {
  .c0 {
    -webkit-flex-basis: 58.3333%;
    -ms-flex-preferred-size: 58.3333%;
    flex-basis: 58.3333%;
    max-width: 58.3333%;
  }
}

@media (min-width:48em) {
  .c0 {
    -webkit-flex-basis: 66.6667%;
    -ms-flex-preferred-size: 66.6667%;
    flex-basis: 66.6667%;
    max-width: 66.6667%;
  }
}

@media screen and (min-width:36em) {
  .c1 {
    padding: 30px 20px;
  }
}

@media screen and (min-width:48em) {
  .c1 {
    padding: 40px 35px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      test
    </h1>
    <div
      className="c3"
    >
      test
    </div>
  </div>
</div>
`;
